lib/driver.js

```diff
@@ -393,8 +393,19 @@ class EspressoDriver extends BaseDriver {
     // start an avd, set the language/locale, pick an emulator, etc...
     // TODO with multiple devices we'll need to parameterize this
     await helpers.initDevice(this.adb, this.opts);
+    this.caps.setAnimation = this.caps.setAnimation || false;
     // https://github.com/appium/appium-espresso-driver/issues/72
-    if (await this.adb.isAnimationOn()) {
+    if (this.caps.setAnimation && await this.adb.isAnimationOn()) {
+      this.log.debug('Animation is already on');
+      this.wasAnimationEnabled = true;
+    } else if (this.caps.setAnimation && (await this.adb.isAnimationOn() === false)) {
+      try {
+        await this.adb.setAnimationState(true);
+        this.wasAnimationEnabled = false;
+      } catch (err) {
+        this.log.warn(`Unable to turn on animations: ${err.message}`);
+      }
+    } else if (this.caps.setAnimation === false && await this.adb.isAnimationOn()) {
```

Comment:
I don't see a need for an explicit `false` equality check after `this.caps.setAnimation = this.caps.setAnimation || false` has been done above. 

Also, is it really necessary to set it explicitly to false?

Reply:
refactor, the state is true unless `disableWindowAnimation` is explicitly set to false.